services/apps/alcs/src/alcs/notice-of-intent/notice-of-intent.service.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  ServiceNotFoundException,
+  ServiceValidationException,
+} from '@app/common/exceptions/base.exception';
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import {
+  FindOptionsRelations,
+  FindOptionsWhere,
+  IsNull,
+  Not,
+  Repository,
+} from 'typeorm';
+import { ApplicationService } from '../application/application.service';
+import { Board } from '../board/board.entity';
+import { CardService } from '../card/card.service';
+import {
+  CreateNoticeOfIntentDto,
+  NoticeOfIntentDto,
+} from './notice-of-intent.dto';
+import { NoticeOfIntent } from './notice-of-intent.entity';
+
+@Injectable()
+export class NoticeOfIntentService {
+  private DEFAULT_RELATIONS: FindOptionsRelations<NoticeOfIntent> = {
+    card: {
+      board: true,
+      type: true,
+      status: true,
+      assignee: true,
+    },
+    localGovernment: true,
+    region: true,
+  };
+
+  constructor(
+    private cardService: CardService,
+    @InjectRepository(NoticeOfIntent)
+    private repository: Repository<NoticeOfIntent>,
+    @InjectMapper() private mapper: Mapper,
+    private applicationService: ApplicationService,
+  ) {}
+
+  async create(data: CreateNoticeOfIntentDto, board: Board) {
+    const existingNoticeOfIntent = await this.repository.findOne({
+      where: {
+        fileNumber: data.fileNumber,
+      },
+    });
+
+    //TODO: Also Check Covenants?
```

Comment:
Is this for future ticket?

Reply:
No, this is for me to ask Mekthi what we think we should when he is back, I'm not quite sure on the best approach.

Reply:
discussed with Daniel the approach, will be a service that will be used for all entities that require the file number to be unique.

Reply:
Going to do this in a future MR.